src/main/kotlin/io/github/corbym/dokker/dokker.kt

```diff
@@ -9,9 +10,48 @@ fun dokker(init: DokkerContainerBuilder.() -> Unit): DokkerContainer {
     return DokkerContainerBuilder().apply(init).build()
 }
 
+// Search order is
+// 1. process name set in code (if set, DokkerAutoProcessSearchResult.processName is not used)
+// 2. environment variable (if found in path)
+// 3. hardcoded "docker" (if found in path)
+// 3. hardcoded "podman" (if found in path)
+object DokkerAutoProcessSearchResult {
+    private fun checkEnvVar(): String? = System.getenv("DOKKER_PROCESS")
+
+    // Note that "command -v" - this is best-effort as adding an 'alias' will cause issues
+    // MacOS defaults to zsh shell.
+    //   Intellij does not read ~/.zprofile nor ~/.zshrc scripts so set environment variables in ~/.profile
+    //   Otherwise, "command -v" may not find the executable
+    private fun processFullPath(process: String): String? {
+        return try {
+            "command -v $process".runCommand()
+        } catch (e: Exception) {
+            println(e)
+            null
+        }
+    }
+
+    // This name is in an object so that we do this once per process, as this check is expensive
+    val processName: String = run {
+        val configured = checkEnvVar()
+        try {
+            listOfNotNull(configured, "docker", "podman").firstNotNullOf { processFullPath(it) }
+        } catch (ex: NoSuchElementException) {
```

Comment:
also not a fan of try catch here, there is a 'magic spell' of `firstNotNullOfOrNull { ... }` 😁 and then you can use `?: error(...)`